Add caution for dag-only deploys and for ci/cd (#2567)
astronomer/docs: astro/deploy-dags.md

## Enable DAG-only deploys on a Deployment

Before you enable DAG-only deploys on a Deployment, ensure the following:

- You have access to the latest version of your Deployment's Astro project.
- You can update your Deployment using the Astro CLI. 
- Your Deployment does not have [CI/CD enforcement](configure-deployment-resources.md#enforce-cicd-deploys) enabled. You can confirm this from the Cloud UI or by running [`astro deployment inspect`](cli/astro-deployment-inspect.md) command.

> :::warning be careful

Reviewer: Always good to do a last-minute review of the preview site before merging. Because we missed the newline here, the note didn't render properly.
![Screenshot 2023-07-17 at 10 19 01 AM](https://github.com/astronomer/docs/assets/74574233/0142ad66-e1e9-4d60-aa8c-8ce9e17a2db5)

## Enable DAG-only deploys on a Deployment

Before you enable DAG-only deploys on a Deployment, ensure the following:

> - You have access to the Deployment's Astro project.

Reviewer: ```suggestion
- You have access to the latest version of your Deployment's Astro project.
```

> - You can trigger deploys from your current machine with the Astro CLI.

Reviewer: ```suggestion
- You can update your Deployment using the Astro CLI. 
```

> - Your Deployment does not have [CI/CD enforcement](configure-deployment-resources.md#enforce-cicd-deploys) enabled. You can also verify this using [Astro CLI's inspect](cli/astro-deployment-inspect.md) command.

Reviewer: ```suggestion
- Your Deployment does not have [CI/CD enforcement](configure-deployment-resources.md#enforce-cicd-deploys) enabled. You can confirm this from the Cloud UI or by running [`astro deployment inspect`](cli/astro-deployment-inspect.md) command.
```

> 
> You cannot enable the DAG-only deploy feature in the Cloud UI.
> 

Reviewer: ```suggestion

```

Reviewer: If there's only one way to do this, we don't need to say they can't do it another way. What would a user do with this information?

Author: @jwitz this was already existing in the original instructions. - "Before you enable DAG-only deploys on a Deployment, ensure that you have access to the Deployment's Astro project and can trigger deploys from your current machine with the Astro CLI. You cannot enable the DAG-only deploy feature in the Cloud UI."

> :::caution 

Author: By looking at "dangerous operation" a user might think it is dangerous to enable dag-only deploys. I would rather suggest to have something like "Be careful" or just default "Danger"

> You must trigger a DAG-based deploy to your Astro Deployment using `astro deploy --dags` immediately after you enable the DAG-only deploy feature. This will ensure your DAGs are refreshed in your Airflow UI using the new deployment method.

Reviewer: When writing a warning, always frame it as "If you don't do this, something bad will happen". Gets people's attention!

> You must trigger a DAG-based deploy to your Astro Deployment using `astro deploy --dags` immediately after you enable the DAG-only deploy feature. This will ensure your DAGs are refreshed in your Airflow UI using the new deployment method.

Author: Using the word "missing" is not giving a positive sign. I would rather use " ...will not be refreshed..."

> If you have Workspace Admin permissions, you can turn off DAG-only deploys for a Deployment at any time if your organization doesn't benefit from faster deploys or prefers a deployment method that is exclusively based on building and deploying your Astro project as a Docker image. To determine if turning off DAG-only deploy functionality is the right choice for your organization, contact [Astronomer support](https://cloud.astronomer.io/support). 

Reviewer: ```suggestion
If you have Workspace Admin permissions, you can turn off DAG-only deploys for a Deployment at any time. To determine if turning off DAG-only deploy functionality is the right choice for your organization, contact [Astronomer support](https://cloud.astronomer.io/support). 

Before you disable DAG-only deploys on a Deployment, ensure the following:

- You have access to the latest version of your Deployment's Astro project.
- You can update your Deployment using the Astro CLI. 
```
